dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04205MigrateVanityURLToContent.java

```diff
@@ -31,375 +33,420 @@
+                String aliases = site.get("text_area1");
+                for (String value : aliases.split("\n")) {
+                    siteIds.put(value, siteId);
+                }
+            }
+
+        }
+        //here get the id from the map to avoid false/positives
+        return siteIds.get(siteName);
+    }
+
+    /**
+     * This method searches for the dotCMS default Language Id.
+     *
+     * @param dc DotConnect Object
+     * @return The default language Id
+     */
+    private int findDefaultLanguageId(DotConnect dc) throws DotDataException {
+
+        dc.setSQL(SELECT_DEFAULT_LANGUAGE);
+        dc.addParam(Config.getStringProperty("DEFAULT_LANGUAGE_CODE", "en"));
+        dc.addParam(Config.getStringProperty("DEFAULT_LANGUAGE_COUNTRY_CODE", "US"));
+
+        List<Map<String, String>> siteLanguage = dc.loadResults();
+        String siteLang = siteLanguage.get(0).get("id");
+
+        return Integer.parseInt(siteLang);
+    }
+
+    /**
+     * Get the content type inode for Host content type
+     *
+     * @param dc DotConnect Object
+     * @return the inode of the Host content Type
+     */
+    private String findHostContentTypeInode(DotConnect dc) throws DotDataException {
+        dc.setSQL(GET_HOST_STRUCTURE_INODE_QUERY);
+        List<Map<String, String>> host = dc.loadResults();
+
+        return host.get(0).get("inode");
+    }
+
+    /**
+     * Remove the virtual link table constraints
+     */
+    private void dropConstraints() throws DotDataException {
+
+        Connection conn = null;
+        try {
+            conn = DbConnectionFactory.getDataSource().getConnection();
+            conn.setAutoCommit(true);
+
+            List<String> tables = getTablesToDropConstraints();
+            if (tables != null) {
+                boolean executeDrop = true;
+                logTaskProgress("==> Retrieving foreign keys [Drop objects? " + executeDrop + "]");
+                getForeingKeys(conn, tables, executeDrop);
+                logTaskProgress("==> Retrieving primary keys [Drop objects? " + executeDrop + "]");
+                getPrimaryKey(conn, tables, executeDrop);
+                logTaskProgress("==> Retrieving indexes [Drop objects? " + executeDrop + "]");
+                getIndexes(conn, tables, executeDrop);
+                logTaskProgress(
+                        "==> Retrieving default constraints [Drop objects? " + executeDrop + "]");
+                getDefaultConstraints(conn, tables, executeDrop);
+                logTaskProgress(
+                        "==> Retrieving check constraints [Drop objects? " + executeDrop + "]");
+                getCheckConstraints(conn, tables, executeDrop);
+                if (DbConnectionFactory.isMsSql()) {
+                    // for mssql we pass again as we might have index dependencies
+                    getPrimaryKey(conn, tables, executeDrop);
+                }
+            }
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)